polaris-react/src/components/RadioButton/RadioButton.scss

```diff
@@ -18,7 +18,7 @@
       @include no-focus-ring;
       outline: var(--p-border-width-2) solid
         var(--p-color-border-interactive-focus);
-      outline-offset: var(--p-space-05);
+      outline-offset: var(--p-space-025);
```

Comment:
Note `RadioButton` doesn't have an error state. Corrected this offset for focus ring to have a space of 1px to match figma

|Before|After|
|-|-|
|<img width="367" alt="Screenshot 2023-06-09 at 3 18 28 PM" src="https://github.com/Shopify/polaris/assets/20652326/f47b96a6-b484-4318-893f-7186a343ee54">|<img width="364" alt="Screenshot 2023-06-09 at 3 18 37 PM" src="https://github.com/Shopify/polaris/assets/20652326/aeb65193-a6ca-4d3c-90a9-fb1c2771f4c9">|